packages/toolkit/package.json

```diff
@@ -23,7 +23,22 @@
     "access": "public"
   },
   "main": "dist/index.js",
-  "module": "dist/redux-toolkit.esm.js",
+  "module": "dist/redux-toolkit.esm.mjs",
```

Comment:
This is a breaking change, which doesn't seem to be intended. Older tools that rely on `"module"` don't all support `.mjs` unfortunately.

This is unnecessary anyhow; `"module"` has always meant ESM, so you can continue to supply `.js` module without worry to it. All tools will be able to handle it correctly. It's only Node that requires `.[cm]js`

Reply:
So Node.js is not a tool? ;-)

So would an acceptable compromise be to reference the `.esm.js` module in the `module` field and the `.esm.mjs` module in the `exports` field?

Reply:
Eh, the runtime/build tool distinction can be a bit silly at times ;)

Yes, that's really the best way to go about it for now. Of course that can be changed in a major (as shipping the same file twice, just with a different extension is a bit silly), but that ensures no breakage. Nothing will consume `"exports"` without being aware of & supporting `.mjs`

Reply:
That's good to know. Are you aware of any overview somewhere of which tools support which field(s) in `package.json`?

Reply:
Webpack's [Package Exports docs](https://webpack.js.org/guides/package-exports/) are the best I am aware of. Huge wall of text, but covers a good amount of conditions, formats, and which popular tools consume what.

Reply:
How about adding the "module" branch to make it safer for Webpack? In my understanding, it corresponds with the top-level "module" field.

```javascript
    ".": {
      "module": "./dist/redux-toolkit.esm.js",
      "import": "./dist/redux-toolkit.esm.mjs",
      "require": "./dist/index.js"
    },
```

Reply:
Safer in what way? Webpack (as well as every other tool, minus any bugs) can handle `.mjs` within `"exports"` just fine. `"module"` is really only useful if you want to provide a different ESM build to some bundlers vs Node.

Reply:
> Safer in what way?

Webpack behaves slightly differently for `*.mjs` (javascript/esm) and `*.js` (javascript/auto):

- It no longer completes the `.js` extension.
- It no longer looks for the `__esModule` flag.
- It no longer polyfills `global` by default.
- It no longer allows non-ESM metadata like `module.hot`.

They are all subtle but I proposed this because you seem to be careful about breaking changes.

Reply:
That's not "safer" though, just allows usage of older (and looser) ESM-ish builds. Not much of a reason to do that if the package is going to output proper ESM anyways.

If you wanted to still output the ESM-ish builds, and do so via `"exports"` for some reason, that could be a valid solution. I just don't think that would solve any issue here, and would only be an extra entry and output file.

Happy to be wrong though if you have an example situation in mind.